File: CPP/Program/Motorcycle.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"  // Mengikutsertakan definisi Vehicle untuk pewarisan

using namespace std;

class Motorcycle : public Vehicle {  // Mendefinisikan kelas Motorcycle yang merupakan turunan dari kelas Vehicle
private:
    string jenisMotor;  // Menyimpan jenis motor
    double kapasitasTangki;  // Menyimpan kapasitas tangki bahan bakar
public:
    Motorcycle() {}  // Konstruktor default

    // Konstruktor dengan parameter
    Motorcycle(string jenisKendaraan,string platNomor, string merk, string tahunProduksi, string warna, string jenis, double tangki)
        : Vehicle(jenisKendaraan, platNomor, merk, tahunProduksi, warna), jenisMotor(jenis), kapasitasTangki(tangki) {}

    // Getter untuk jenisMotor
    string getJenisMotor() const {
        return jenisMotor;
    }

    // Setter untuk jenisMotor
    void setJenisMotor(string jenisMotor) {
        this->jenisMotor = jenisMotor;
    }

    // Getter untuk kapasitasTangki
    double getKapasitasTangki() const {
        return kapasitasTangki;
    }

    // Setter untuk kapasitasTangki
    void setKapasitasTangki(double kapasitasTangki) {
        this->kapasitasTangki = kapasitasTangki;
    }

    // Metode override untuk menampilkan informasi sepeda motor
    void display() const override {
        Vehicle::display();  // Memanggil metode display dari kelas dasar
        cout << "Jenis Motor        : " << jenisMotor << endl;  // Menampilkan jenis motor
        cout << "Kapasitas Tangki   : " << kapasitasTangki << " Liter" << endl;  // Menampilkan kapasitas tangki
    }

    ~Motorcycle(){  // Destruktor
        // Tidak ada operasi khusus yang diperlukan dalam destruktur ini
    }
};

File: Python/Program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    def __init__(self, jenisKendaraan="", platNomor="", merk="", tahunProduksi="", warna="", jenisMotor="", kapasitasTangki=0.0):
        # Memanggil konstruktor kelas induk (Vehicle)
        super().__init__(jenisKendaraan, platNomor, merk, tahunProduksi, warna)
        # Menginisialisasi atribut khusus motor
        self.jenisMotor = jenisMotor
        self.kapasitasTangki = kapasitasTangki

    def display(self):
        # Memanggil metode display kelas induk (Vehicle)
        super().display()
        # Menampilkan informasi tambahan tentang motor
        print("Jenis Motor        :", self.jenisMotor)
        print("Kapasitas Tangki   :", self.kapasitasTangki, "Liter")

    def getJenisMotor(self):
        # Mengembalikan jenis motor
        return self.jenisMotor

    def setJenisMotor(self, jenisMotor):
        # Mengatur jenis motor
        self.jenisMotor = jenisMotor

    def getKapasitasTangki(self):
        # Mengembalikan kapasitas tangki motor
        return self.kapasitasTangki

    def setKapasitasTangki(self, kapasitasTangki):
        # Mengatur kapasitas tangki motor
        self.kapasitasTangki = kapasitasTangki
